lib/confluence/vendor_specific.es6.js

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+require('./vendor_specific_data.es6.js');
+
+foam.CLASS({
+  package: 'org.chromium.apis.web',
+  name: 'VendorSpecific',
+  extends: 'org.chromium.apis.web.MetricComputer',
+  documentation: `VendorSpecific is a class that computes the vendor-specific
+    API metrics from browserApiDAO. It shows the number of APIs from this
+    browser vendor. These APIs were shipped by a vendor at least 1 year ago,
+    and haven't since been removed (i.e., appear in all of the last year's
+    releases), and have not been visible in any other browser vendor during
+    the last year's releases.`,
+  requires: [
+    'org.chromium.apis.web.Browser',
+    'org.chromium.apis.web.WebInterface',
+    'org.chromium.apis.web.BrowserWebInterfaceJunction',
+    'org.chromium.apis.web.VendorSpecificData',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.dao.ArraySink',
+    'foam.dao.EasyDAO',
+  ],
+  properties: [
+    {
+      name: 'vendorSpecificDAO',
+      documentation: `This is a DAO that contains vendor specific data.`,
+      final: true,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'vendorSpecificDAO',
+          of: this.VendorSpecificData,
+          daoType: 'MDAO',
+        });
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'compute',
+      documentation: `Compute vendor specific API value for each browser in
+          browsers at the given date.`,
+      code: function(browsers, date) {
+        for (let i = 0; i < browsers.length; i++) {
+          let browser = browsers[i];
+          let year = date.getFullYear();
+          let month = date.getMonth();
+          let day = date.getDate();
+          let previousYear = new Date(year - 1, month, day);
+          let browserKeys = browsers.map((browser) => browser.browserKey);
+          // Find all browsers that satisfy one of:
+          //   - a previous version of the browser.
+          //   - some other browser but released in previous one year range.
+          //   - if no release in one year, use most recent browser.
+          this.browserDAO.where(this.mlang.OR(
+              this.mlang.AND(
+                  this.mlang.IN(this.Browser.BROWSER_KEY, browserKeys)),
+              this.mlang.AND(
+                  this.mlang.EQ(this.Browser.BROWSER_NAME, browser.browserName),
+                  this.mlang.EQ(this.Browser.OS_NAME, browser.osName),
+                  this.mlang.GT(this.Browser.RELEASE_DATE, previousYear),
+                  this.mlang.LT(this.Browser.RELEASE_DATE, date)),
+              this.mlang.AND(
+                  this.mlang.GT(this.Browser.RELEASE_DATE, previousYear),
+                  this.mlang.LT(this.Browser.RELEASE_DATE, date))
+              )).select().then((result) => {
+            // browserIface are a map of the form:
```

Comment:
In general, I think that you want to indent a block 2 more than the line containing the start of the block.